core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryAdapter.kt

```diff
@@ -0,0 +1,145 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.history
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import butterknife.BindView
+import butterknife.ButterKnife
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R.layout
+import org.kiwix.kiwixmobile.core.R2.id
+import org.kiwix.kiwixmobile.core.downloader.model.Base64String
+import org.kiwix.kiwixmobile.core.extensions.ViewGroupExtensions.inflate
+import org.kiwix.kiwixmobile.core.extensions.setBitmap
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.DateItem
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.HistoryItem
+import org.threeten.bp.LocalDate
+import org.threeten.bp.format.DateTimeFormatter
+
+internal class HistoryAdapter(
+  val historyList: List<HistoryListItem>,
+  val deleteList: List<HistoryListItem>,
+  val itemClickListener: OnItemClickListener
+) : Adapter<ViewHolder>() {
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+    return if (viewType == TYPE_ITEM) {
+      Item(parent.inflate(layout.item_bookmark_history, false))
+    } else {
+      Category(parent.inflate(layout.header_date, false))
+    }
+  }
+
+  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+    if (holder is Item) {
+      val history = historyList[position] as HistoryItem
+      setItemDataWithHelpOfHistoryItem(holder, history)
+    } else if (holder is Category) {
+      val date = (historyList[position] as DateItem).dateString
+      setCategoryDataWithHelpOfDate(date, holder)
+    }
+  }
+
+  private fun setCategoryDataWithHelpOfDate(date: String, holder: Category) {
+    val todaysDate = LocalDate.now()
+    val yesterdayDate = LocalDate.now().minusDays(1)
+    val formatter = DateTimeFormatter.ofPattern("d MMM yyyy")
+    val givenDate = LocalDate.parse(date, formatter)
+
+    if (todaysDate?.equals(givenDate) == true) {
+      holder.date.setText(
+        R.string.time_today
+      )
+    } else if (yesterdayDate?.equals(givenDate) == true) {
+      holder.date.setText(
+        R.string.time_yesterday
+      )
+    } else {
+      holder.date.text = date
+    }
+  }
+
+  fun ImageView.setImageDrawableCompat(context: Context, id: Int) {
+    setImageDrawable(ContextCompat.getDrawable(context, id))
```

Comment:
this is an extension on ImageView which means `this` inside this scope is the ImageView that invokes `setImageDrawableCompat` and all views have a context so you don't need tcontext in the method parameters.

This could also be moved to ImageViewExtensions